Index: LinearRing.js
===================================================================
--- LinearRing.js	(revision 10488)
+++ LinearRing.js	(working copy)
@@ -265,7 +265,15 @@
             for(var i=0; i<len-1; i++) {
                 p1 = ring.components[i];
                 p2 = ring.components[i+1];
-                area += OpenLayers.Util.rad(p2.x - p1.x) *
+                var d = p2.x - p1.x;
+                if (Math.abs(d) > 180) {
+                  if (d>0) {
+                    d += -360
+                  } else {
+                    d += 360
+                  }
+                }
+                area += OpenLayers.Util.rad(d) *
                         (2 + Math.sin(OpenLayers.Util.rad(p1.y)) +
                         Math.sin(OpenLayers.Util.rad(p2.y)));
             }
